Services/Services/ProjectCategoryService.cs

```diff
@@ -0,0 +1,118 @@
+*
+* You can find a copy of the GNU Lesser General Public License
+* along with this program, in the LICENSE.md file in the root project directory.
+* If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
+*/
+
+using Models;
+using Repositories;
+using Services.Base;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Services.Services
+{
+
+    /// <summary>
+    ///     This is the project category service interface
+    /// </summary>
+    public interface IProjectCategoryService : IService<ProjectCategory>
+    {
+        /// <summary>
+        ///     Gets project category by given project and category id
+        /// </summary>
+        Task<ProjectCategory> GetProjectCategory(int projectId, int categoryId);
+
+        /// <summary>
+        ///     Gets project category by given category id
+        /// </summary>
+        Task<ProjectCategory> GetProjectCategory(int categoryId);
+
+        /// <summary>
+        ///     Update project categories by given project and categories
+        /// </summary>
+        Task UpdateProjectCategories(Project project, IEnumerable<Category> categories);
+
+        /// <summary>
+        ///     Clear project categories by given project
+        /// </summary>
+        Task ClearProjectCategories(Project project);
+    }
+
+    /// <summary>
+    ///     This is the project category service
+    /// </summary>
+    public class ProjectCategoryService : Service<ProjectCategory>,
+                                          IProjectCategoryService
+    {
+
+        /// <summary>
+        ///     This is the project category constructor
+        /// </summary>
+        /// <param name="repository"></param>
+        public ProjectCategoryService(IProjectCategoryRepository repository) :
+            base(repository) { }
+
+        /// <summary>
+        ///     Gets the repository
+        /// </summary>
+        private new IProjectCategoryRepository Repository =>
+            (IProjectCategoryRepository) base.Repository;
+
+        /// <summary>
+        ///     Gets project category by given project and category id
+        /// </summary>
+        public Task<ProjectCategory> GetProjectCategory(int projectId, int categoryId)
+        {
+            return Repository.GetProjectCategory(projectId, categoryId);
+        }
+
+        /// <summary>
+        ///     Gets project category by given category id
+        /// </summary>
+        public Task<ProjectCategory> GetProjectCategory(int categoryId)
+        {
+            return Repository.GetProjectCategory(categoryId);
+        }
+
+        /// <summary>
+        ///     Update project categories by given project and categories
+        /// </summary>
+        public async Task UpdateProjectCategories(Project project, IEnumerable<Category> categories)
+        {
+            IEnumerable<ProjectCategory> allProjectCategories = await Repository.GetAll();
```

Comment:
I think its better to use a get by projectId here instead of loading all ProjectCategories in memory

Reply:
Nuances.. Maybe for a next issue?